src/utils/validate-schema.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Checks if Valid default coordinates
+ * @param {unknown} possibleCoordinates
+ * @param {number} dimensions - 2 for 2D Vector or 3 for 3D vector
+ * @returns {boolean} A boolean determining if coordinates are parsed correctly.
+ */
+function isValidDefaultCoordinate (possibleCoordinates, dimensions) {
+  if (typeof possibleCoordinates !== 'object' || possibleCoordinates === null) {
+    return false;
+  } else if (Object.keys(possibleCoordinates).length !== dimensions) {
+    return false;
+  } else {
+    if (dimensions === 2 && (possibleCoordinates.x === 0 || possibleCoordinates.x) && (possibleCoordinates.y === 0 || possibleCoordinates.y)) {
```

Comment:
I think we could condense this code more. We only need to check X/Y/Z/W each once.